unit_tests/test_events.py

```diff
@@ -338,3 +340,31 @@ def test_failed_stall_during_filter(self):
                                line="[99.80.124.204] [stdout] Mar 31 09:08:10 warning|  reactor stall 20").publish()
 
         self.assertEqual(event.severity, Severity.DEBUG)
+
+    def test_duration_format(self):
+        event = NodetoolEvent(nodetool_command="scrub --skip-corrupted drop_table_during_repair_ks_0", node='1.0.0.121',
+                              options="more options", publish_event=False)
+
+        event.duration = 2.52
+        self.assertEqual(event.duration_formatted, '2.52s')
+
+        event.duration = 326
+        self.assertEqual(event.duration_formatted, '5m26s')
+
+        event.duration = 4598
+        self.assertEqual(event.duration_formatted, '1h16m38s')
+
+        event.duration = 87400
+        self.assertEqual(event.duration_formatted, '1d0h16m40s')
+
+    @mock.patch('sdcm.sct_events.base.SctEvent.publish')
+    def test_publish(self, publish):
+        event = NodetoolEvent(nodetool_command="scrub", node='1.0.0.121',
+                              options="", publish_event=True)
+        event.begin_event()
+        self.assertTrue(publish.called, "Publish function was not called unexpectedly")
+
+        publish.reset_mock()
+        event.publish_event = False
+        event.begin_event()
+        self.assertFalse(publish.called, "Publish function was called unexpectedly")
```

Comment:
2 unit tests are mixed into one here. Need to split these.
Must be something like "test_publish_false" and "test_publish_true".

Reply:
I didn't want to split because in this case we need to initialise the event twice